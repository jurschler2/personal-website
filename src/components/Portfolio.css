/** 
 * Styling for the Portfolio component:
 */

 .portfolio-header {
  font-family: Kanit;
  font-weight: 300;
  font-size: 115%;
  color: #7b7b87;
  display:  block;
  margin-left: 10%;
  width: 80%;
  text-align: left;
  margin-bottom: 5%;
  margin-top: 2%;
 }


 @media screen and (min-width: 760px) {
 .portfolio-header {
  font-family: Kanit;
  font-weight: 300;
  font-size: calc(.5rem + 1.5vw);
  color: #7b7b87;
  display:  block;
  margin-left: 10.5%;
  text-align: left;
  margin-bottom: 5%;
  margin-top: 2%;
 }
}