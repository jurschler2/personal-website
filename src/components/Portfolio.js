import React from "react";
import ProjectCard from "./ProjectCard"
import { projects } from "../content/data"
import { Container, Row, Col } from "reactstrap";

/**
 *  DESCRIPTION:
 *  PROPS:
 *  FLOW: App => Routes => Portfolio
 *  PARENT: Routes
 *  CHILDREN: ProjectCard
 */

function Portfolio() {

  const renderProjects = () => {

    return projects.map(p => (
                              <ProjectCard title={p.title} 
                                           description={p.description} 
                                           gallery={p.gallery}
                                           link={p.link}
                                           tech={p.tech} />
                             )
                       )
  }

  return (
   <Container className="sectionContainer" id="/portfolio"> 
   <Row>
     <Col md={12} lg={12}>
       <div className="portfolio-header">
          <h3 data-aos="fade-down" className="text-center">
            Portfolio
          </h3>
          <div data-aos="fade-right" className="portfolioRule mx-auto mt-3 mb-5"></div>
       </div>
     </Col>
   
    <Col md={12} lg={12}>

    <div className="projectCardContainer">
      {renderProjects()}
    </div> 
    </Col>
   </Row>
   </Container>
  )
}

export default Portfolio;