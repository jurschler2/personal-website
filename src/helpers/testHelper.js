import React from "react";
import { render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

export function renderWithRouter(children, initialPath = "") {
  const history = createMemoryHistory();
  if (initialPath) history.push(initialPath);
  return render(<Router history={history}>{children}</Router>);
}
