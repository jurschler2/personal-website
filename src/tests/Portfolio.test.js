import React from "react";
import Portfolio from "../components/Portfolio";
import {renderWithRouter} from "../helpers/testHelper"

it("renders without crashing", () => {
  renderWithRouter(<Portfolio />);
});

it("matches snapshot", () => {
  const { asFragment } = renderWithRouter(<Portfolio />);
  expect(asFragment()).toMatchSnapshot();
});