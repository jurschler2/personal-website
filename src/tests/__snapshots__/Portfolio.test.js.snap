// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="portfolio-header"
  >
    Here are some things I have built:
  </div>
  <div
    class="projectcard-container"
  >
    <div
      class="project-container"
    >
      <div
        class="project-img-container"
      >
        <div
          class="carousel slide"
        >
          <ol
            class="carousel-indicators"
          >
            <li
              class="active"
            />
            <li
              class=""
            />
            <li
              class=""
            />
            <li
              class=""
            />
          </ol>
          <div
            class="carousel-inner"
          >
            <div
              class="carousel-item active"
            >
              <img
                alt="Image 1"
                src="microblog_1.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  Landing Page
                </strong>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Image 2"
                src="microblog_2.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  Add a Post
                </strong>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Image 3"
                src="microblog_3.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  Post Details and Add Comments
                </strong>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Image 4"
                src="microblog_4.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  Edit a Post
                </strong>
              </div>
            </div>
          </div>
          <a
            class="carousel-control-prev"
            role="button"
            style="cursor: pointer;"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-prev-icon"
            />
            <span
              class="sr-only"
            >
              Previous
            </span>
          </a>
          <a
            class="carousel-control-next"
            role="button"
            style="cursor: pointer;"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-next-icon"
            />
            <span
              class="sr-only"
            >
              Next
            </span>
          </a>
        </div>
      </div>
      <div
        class="project-description-container"
      >
        <div
          class="project-title"
        >
          <strong>
            Microblog React App
          </strong>
        </div>
        <div
          class="project-description"
        >
          <p>
            Microblog is a full-stack React application that is a basic blogging site. Anyone is able to go to the site, create posts, vote on posts, and leave comments on posts. The primary focus of this project was state management through the usage of React Redux. The rest of the frontend was built out using React. The backend was written Node/Express, and handles calls from the frontend to update post data within a PostgreSQL database. This project was my first time using Redux for more robust state management, and it helped me understand just how useful Redux could be, especially for an application that should be scalable with many posts/comments/etc. Check it out on 
            <a
              href="https://github.com/jurschler2/react-redux-microblog"
            >
              Github
            </a>
            .
          </p>
        </div>
      </div>
    </div>
    <div
      class="project-container"
    >
      <div
        class="project-img-container"
      >
        <div
          class="carousel slide"
        >
          <ol
            class="carousel-indicators"
          >
            <li
              class="active"
            />
            <li
              class=""
            />
            <li
              class=""
            />
            <li
              class=""
            />
            <li
              class=""
            />
            <li
              class=""
            />
            <li
              class=""
            />
            <li
              class=""
            />
          </ol>
          <div
            class="carousel-inner"
          >
            <div
              class="carousel-item active"
            >
              <img
                alt="Image 1"
                src="jobly_1.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  Landing Page
                </strong>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Image 2"
                src="jobly_2.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  User Sign-Up
                </strong>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Image 3"
                src="jobly_3.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  User Login
                </strong>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Image 4"
                src="jobly_4.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  Homepage
                </strong>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Image 5"
                src="jobly_5.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  Companies
                </strong>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Image 6"
                src="jobly_6.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  Apply to a Company's Jobs
                </strong>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Image 7"
                src="jobly_7.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  Apply to Any Jobs
                </strong>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Image 8"
                src="jobly_8.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  User's Profile
                </strong>
              </div>
            </div>
          </div>
          <a
            class="carousel-control-prev"
            role="button"
            style="cursor: pointer;"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-prev-icon"
            />
            <span
              class="sr-only"
            >
              Previous
            </span>
          </a>
          <a
            class="carousel-control-next"
            role="button"
            style="cursor: pointer;"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-next-icon"
            />
            <span
              class="sr-only"
            >
              Next
            </span>
          </a>
        </div>
      </div>
      <div
        class="project-description-container"
      >
        <div
          class="project-title"
        >
          <strong>
            Jobly React App
          </strong>
        </div>
        <div
          class="project-description"
        >
          <p>
            Jobly is a full-stack React application that enables users to apply for jobs at specific companies. To view the job listings, users must sign up and log in to receive a login token for authentication. Nearly all routes in this site are protected by a middleware authentication function that validates the login token provided by the user. Valid users are able to search for jobs, apply for jobs, and edit their profiles. This was one of my first React projects, and it helped me learn the usefulness of components and hooks, as well as frontend hierarchical design. Check it out on 
            <a
              href="https://github.com/jurschler2/react-jobly"
            >
              Github
            </a>
            .
          </p>
        </div>
      </div>
    </div>
    <div
      class="project-container"
    >
      <div
        class="project-img-container"
      >
        <div
          class="carousel slide"
        >
          <ol
            class="carousel-indicators"
          >
            <li
              class="active"
            />
            <li
              class=""
            />
            <li
              class=""
            />
            <li
              class=""
            />
            <li
              class=""
            />
            <li
              class=""
            />
            <li
              class=""
            />
            <li
              class=""
            />
          </ol>
          <div
            class="carousel-inner"
          >
            <div
              class="carousel-item active"
            >
              <img
                alt="Image 1"
                src="warbler_1.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  Landing Page
                </strong>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Image 2"
                src="warbler_2.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  User Sign-Up
                </strong>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Image 3"
                src="warbler_3.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  User Login
                </strong>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Image 4"
                src="warbler_4.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  Homepage
                </strong>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Image 5"
                src="warbler_5.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  User Profile
                </strong>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Image 6"
                src="warbler_6.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  User's Message
                </strong>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Image 7"
                src="warbler_7.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  User Create Message
                </strong>
              </div>
            </div>
            <div
              class="carousel-item"
            >
              <img
                alt="Image 8"
                src="warbler_8.png"
              />
              <div
                class="carousel-caption"
              >
                <strong>
                  Search Users
                </strong>
              </div>
            </div>
          </div>
          <a
            class="carousel-control-prev"
            role="button"
            style="cursor: pointer;"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-prev-icon"
            />
            <span
              class="sr-only"
            >
              Previous
            </span>
          </a>
          <a
            class="carousel-control-next"
            role="button"
            style="cursor: pointer;"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-next-icon"
            />
            <span
              class="sr-only"
            >
              Next
            </span>
          </a>
        </div>
      </div>
      <div
        class="project-description-container"
      >
        <div
          class="project-title"
        >
          <strong>
            Warbler Flask App
          </strong>
        </div>
        <div
          class="project-description"
        >
          <p>
            Warbler is a full-stack Flask application that functions like a lite clone of Twitter. To interact with the site, a user must first sign up and then login to receive a login token for authentication. Nearly all functionality is protected by a python decorator in the Flask backend to authenticate a user. Once inside the application, users can follow other users, write warbles (tweets), view follower or all warbles, like warbles, among other things. This project was written with a Jinja-supplied frontend, Flask backend, and a PostgreSQL database. The focus of this project was primarily on the database structure and table schemas to map out relationships between users, warbles, etc. SQLAlchemy was used in conjunction with the Flask routes to abstract user specific data based on the route requested. At the time, this was my biggest Python (as well as SQL) project, and it helped me get a better understanding of backend routing and database design patterns. Check it out on 
            <a
              href="https://github.com/jurschler2/flask-warbler"
            >
              Github
            </a>
            .
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
